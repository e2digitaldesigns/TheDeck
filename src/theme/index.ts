export * from "./styles/forms.styles";
export * from "./styles/scrollBars.styles";
