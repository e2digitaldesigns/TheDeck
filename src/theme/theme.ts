export const theme = {
  sizes: {
    application: {
      height: "785px",
      width: "1280px"
    },
    header: {
      height: "2rem",
      width: "100vw"
    },
    macroDeck: {
      sideBar: { width: "240px" },
      mainContent: { width: "1016px" }
    }
  },
  colors: {
    body: "black",
    border: "#111",
    accent1: "#f542e9",
    accent2: "#291697",
    accent3: "#404040",
    accent4: "#2b2b2b",
    accent5: "#1a1a1a",
    gridBg: "43, 45, 48,",
    font: "white",
    main: "#100727",
    mainBackground: "black",
    scrollBackground: "#100727",
    scrollFont: "#dedede",
    scrollHeader: "black",
    scrollHeaderFont: "white"
  }
};
