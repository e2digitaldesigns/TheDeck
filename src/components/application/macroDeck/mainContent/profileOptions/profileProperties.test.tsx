import React from "react";
import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";
import ProfileProperties from "./profileProperties";

import { ThemeProvider } from "styled-components";
import { theme } from "../../../../../theme/theme";
import * as hooks from "../../../../../hooks";

const mocked = {
  readProfile: jest.fn(),
  updateProfile: jest.fn()
};

const mockData = {
  profile: {
    _id: "profile-id-001",
    profileName: "001 profile 6 pads",
    buttonPads: 6
  }
};

jest.mock("../../../../../hooks", () => ({
  useProfile: () => ({
    readProfile: () => mockData.profile,
    updateProfile: mocked.updateProfile
  })
}));

const testSetup = () => {
  return render(
    <ThemeProvider theme={theme}>
      <ProfileProperties profileId={mockData.profile._id} />
    </ThemeProvider>
  );
};

let wrapper: any = null;
beforeEach(() => {
  wrapper = testSetup();
});

afterEach(() => {
  wrapper = null;
  jest.clearAllMocks();
});

describe("<MacroDeck/>", () => {
  it("Should render without errors", () => {
    const component = wrapper;
    expect(component).toMatchSnapshot();
  });

  it("Should accept profile name onChange Event", () => {
    const component = wrapper;
    const input = component.getByTestId("profile_props__profile-name");
    expect(input).toBeTruthy();

    const goodName = "Profile 001";
    fireEvent.change(input, { target: { value: goodName } });
    expect(input.value).toBe(goodName);

    fireEvent.change(input, { target: { value: "?lo()lo*" } });
    expect(input.value).toBe("lolo");
  });

  it("Should accept buttonPad onChange Event", () => {
    const component = wrapper;
    const select = component.getByTestId("profile_props__button-pads");
    expect(select).toBeTruthy();

    fireEvent.change(select, { target: { value: "15" } });
    expect(select.value).toBe("15");
  });

  it("Should accept buttonPad onChange Event", () => {
    const component = wrapper;
    const submit = component.getByTestId("profile_props__submit");
    expect(submit).toBeTruthy();

    const newData = {
      ...mockData.profile,
      profileName: "Boogie",
      buttonPads: 15
    };

    fireEvent.change(component.getByTestId("profile_props__profile-name"), {
      target: { value: newData.profileName }
    });

    fireEvent.change(component.getByTestId("profile_props__button-pads"), {
      target: { value: newData.buttonPads }
    });

    fireEvent.click(submit);

    expect(mocked.updateProfile).toHaveBeenCalledTimes(1);
    expect(mocked.updateProfile).toHaveBeenCalledWith(newData._id, newData);
  });
});
