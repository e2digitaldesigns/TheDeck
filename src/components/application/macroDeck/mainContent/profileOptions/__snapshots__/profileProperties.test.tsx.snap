// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MacroDeck/> Should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <input
          class="sc-gsDKAQ sc-jRQBWg dEgDrl gVIDSv"
          data-testid="profile_props__profile-name"
          type="text"
          value="001 profile 6 pads"
        />
        <select
          class="sc-dkPtRN sc-gKclnd iRRDCl lotgiN"
          data-testid="profile_props__button-pads"
        >
          <option
            value="6"
          >
            Button Pads: 6
          </option>
          <option
            value="8"
          >
            Button Pads: 8
          </option>
          <option
            value="12"
          >
            Button Pads: 12
          </option>
          <option
            value="15"
          >
            Button Pads: 15
          </option>
          <option
            value="24"
          >
            Button Pads: 24
          </option>
          <option
            value="32"
          >
            Button Pads: 32
          </option>
        </select>
        <button
          class="sc-hKwDye sc-iCfMLu PLLBx bRXAck"
          data-testid="profile_props__submit"
        >
          Submit
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <input
        class="sc-gsDKAQ sc-jRQBWg dEgDrl gVIDSv"
        data-testid="profile_props__profile-name"
        type="text"
        value="001 profile 6 pads"
      />
      <select
        class="sc-dkPtRN sc-gKclnd iRRDCl lotgiN"
        data-testid="profile_props__button-pads"
      >
        <option
          value="6"
        >
          Button Pads: 6
        </option>
        <option
          value="8"
        >
          Button Pads: 8
        </option>
        <option
          value="12"
        >
          Button Pads: 12
        </option>
        <option
          value="15"
        >
          Button Pads: 15
        </option>
        <option
          value="24"
        >
          Button Pads: 24
        </option>
        <option
          value="32"
        >
          Button Pads: 32
        </option>
      </select>
      <button
        class="sc-hKwDye sc-iCfMLu PLLBx bRXAck"
        data-testid="profile_props__submit"
      >
        Submit
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
